flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/delegation/DialectFactory.java

```diff
@@ -36,52 +34,22 @@
  * <p>The {@link #factoryIdentifier()} is identified by matching it against {@link
  * TableConfigOptions#TABLE_SQL_DIALECT}.
  */
-@Internal
+@PublicEvolving
 public interface DialectFactory extends Factory {
-
     /** Creates a new parser. */
     Parser create(Context context);
 
+    /** Create an extended operation executor. */
     default ExtendedOperationExecutor createExtendedOperationExecutor(Context context) {
```

Comment:
I'm thinking of removing the `ExtendedOperationExecutor` because we have introduced `ExecutableOperation` which is a self-contained way to support the operation execution. In this way, we can rename `DialectFactory` back to `ParserFactory` to align with the original FLIP proposal.

Reply:
Removing `ExtendedOperationExecutor` is good idea. It makes everything simple. I have updated it for removing and renming.
Hope the internal `ExecutableOperation` interface won't change frequently